app/src/utils/validate.js

```diff
@@ -55,4 +55,47 @@ export function getQSLObjTypesAndProjection(query) {
     }
   });
   return queryProjection;
-}
\ No newline at end of file
+}
+
+//TODO:DM - figure out how to extract shared code between following two functions into another helper fn which both can use
+export const addResourceIdFilterQSL = (query = '', resourceId = '') => {
```

Comment:
ok for now. it is better to always return the resourceid from the API side as it suppose to be the document id.